### DISCUSSION TITLE
[react-native] Push notifications in WalletConnect 2.0

### DISCUSSION THREAD

**Discussion by vm-agnislav** [WalletConnect/walletconnect-monorepo#3075] [Developer Support] [answered]
Hi there. 

We've got an RN app and there was a push notification support in v1. However, after migration to v2 we aren't able to receive push notifications anymore. There are instructions for [Android](https://docs.walletconnect.com/2.0/android/push/wallet-usage/getting-started) and [iOS](https://docs.walletconnect.com/2.0/ios/push/wallet-usage/getting-started) but literally nothing for [React Native](https://docs.walletconnect.com/2.0/reactnative/web3wallet/wallet-usage). 

What is the expectation for them?
- They're not available and should be set up using the native platforms instructions?
- They're available but the doc isn't up-to-date? If so, could you provide anything so we can implement it?
- They're not available but somewhere in the roadmap?

Best Regards,
Agnislav.

**Comment by crypblizz8** [comment]
@Cali93 @devceline 👀

**Comment by Cali93** [comment]
Hey @vm-agnislav,

We are working on a React Native SDK.
We will let you know as soon as it's available.

**Reply by vm-agnislav** [reply]
@Cali93 awesome, thanks for the quick response. Are there any estimates to have them ready? i.e. week/month/3 months?

**Reply by Cali93** [reply, marked as answer]
I'd say best case estimate ~1 month and worst case 2 or 3 months.

**Comment by mwaleed-39** [comment]
I faced some issue in react native. when we pressed upload image, the error
message shows:

How to call this API, when we upload image in front end side, the image
will be stored in mongodb
I create API:
app.post('/finerequest', (req, res, next) => {

const file = req.files.image;
cloudinary.uploader.upload(file.tempFilePath, (err, result) => {
console.log(result)
const requestId = req.body.request_id
const fine = new Fine({
_id: new mongoose.Types.ObjectId,
request_id: requestId,
image: result.url
})
fine.save()
.then(result => {
res.status(200).json({
message: "Fine and Vehicle Repair Request added",
response: result
})
})
.catch(err => {
res.status(500).json({
message: "Fine and Vehicle request not added",
error: err
})
})
})
})

CODE:

import React, { useState } from 'react'
import { Alert, Button, View,StyleSheet } from 'react-native'
import * as ImagePicker from 'react-native-image-picker'
import {launchImageLibrary} from 'react-native-image-picker'

const options ={
title: 'Select Image',
type: 'library',
options: {
maxHeight:200,
maxWidth:200,
selectionLimit: 1,
mediaType: 'photo',
includeBase64: false,
}
}

const Request = () => {
const opengallery = async()=>{
const image = await ImagePicker.launchImageLibrary(options);
console.log(image)
}

return (

)
}
export default Request;

const styles = StyleSheet.create({
container:{
flex:1,
alignItems:'center',
justifyContent:'center'
}
})

when we pressed upload button, the error message shows:

Possible Unhandled Promise Rejection (id: 1):
TypeError: Cannot read property 'launchImageLibrary' of null
TypeError: Cannot read property 'launchImageLibrary' of null

Again and again same error show, we Image picker library, I think it has
been deprecated

**Reply by Cali93** [reply]
This is unrelated to WalletConnect and looks like an issue with cloudinary. Please open an issue on their repository.
